ccdproc/shutter_correction.py

```diff
@@ -0,0 +1,347 @@
+        Use the `Surma1993` algorithm to determine the shutter map instead of
+        the `GaladiEnriqez1995`.
+
+    min_files_to_sigma_clip : `int`, optional, defaults to 5
+        The `min_files_to_sigma_clip` parameter sets the minimum
+        number of files at each exposure time that must exist before a sigma
+        clipping algorithm is used in the combination.  Otherwise the combine is
+        simply an average combine.
+
+    sigma_clip_low_thresh : `int`, optional, defaults to 5
+    sigma_clip_high_thresh : `int`, optional, defaults to 5
+        These are the sigma clipping thresholds which are used in the combining
+        of files (if the `min_files_to_sigma_clip` threshold is exceeded).  They
+        are passed directly to the `ccdproc.combine` task.
+
+    normalizer : `function`, optional defaults to `lambda flat: flat.data.max()`
+        This is the normalizer function used by the `Surma1993` algorithm.
+        This is ignored if the `GaladiEnriqez1995` default algorithm is used.
+
+    Returns
+    -------
+    shutter_map : `~ccdproc.CCDData`
+        The shutter correction map in units of seconds.
+
+    '''
+    bytype = files.summary.group_by(image_type_keyword)
+    flats = bytype.groups[bytype.groups.keys[image_type_keyword]\
+                          == flat_keyword_value]
+    biases = bytype.groups[bytype.groups.keys[image_type_keyword]\
+                           == bias_keyword_value]
+
+    bias_images = [fits_ccddata_reader(os.path.join(files.location, f))
+                   for f in biases['file']]
+    if len(bias_images) > 0:
+        ## Generate master bias to subtract from each flat
+        if len(bias_images) > 1:
+            master_bias = combine(bias_images, combine='median')
+        else:
+            master_bias = bias_images[0]
+
+    assert exptimekey in flats.keys()
+    flats = flats.group_by(exptimekey)
+    combined_flats = []
+    for exptime in flats.groups.keys[exptimekey]:
+        thisexptime = flats.groups[flats.groups.keys[exptimekey] == exptime]
+        ## Combine each group of flats with same exposure times
+        flat_images = [fits_ccddata_reader(os.path.join(files.location, f))
+                       for f in thisexptime['file']]
+        if len(flat_images) > 1:
+            # Bias subtract flats
+            if len(biases) > 0:
+                flat_images = [subtract_bias(im, master_bias) for im in flat_images]
+            sigma_clip = len(flat_images) > min_files_to_sigma_clip
+            combined_flat = combine(flat_images, method='average',
+                                    sigma_clip=sigma_clip,
+                                    sigma_clip_low_thresh=sigma_clip_low_thresh,
+                                    sigma_clip_high_thresh=sigma_clip_high_thresh)
+            combined_flats.append(combined_flat)
+            assert combined_flat.shape == combined_flats[0].shape
+        else:
+            combined_flats.append(flat_images[0])
+
```

Comment:
Can you break the steps below into their own task.   Users might want to have there own way to create the biases or flats and not the way that you have outlined here.   Just have a task that the combined bias or flat field is passed to would provide the users more flexibility and seems like a better place to split the two steps.